src/lightning/fabric/strategies/fsdp.py

```diff
@@ -886,3 +891,8 @@ def _has_meta_device_parameters(obj: Union[Module, Optimizer]) -> bool:
     if isinstance(obj, Module):
         return any(t.is_meta for t in obj.parameters())
     raise TypeError(f"Expected `torch.nn.Module` or `torch.optim.Optimizer`, got: {type(obj).__name__}")
+
+
+def _move_torchmetrics_to_device(module: torch.nn.Module, device: torch.device) -> None:
+    for metric in (m for m in module.modules() if isinstance(m, Metric)):
+        metric.to(device)
```

Comment:
`.to()` is inplace for metrics? Maybe worth leaving a comment for those who think assignment is missing

Reply:
Yep. It's generally the case for nn.Modules, but I've added the comment.